sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -2007,4 +2008,125 @@ class RegexUnsupportedException(message: String, index: Option[Int])
       case _ => message
     }
   }
+}
+
+sealed trait RegexOptimizationType
+object RegexOptimizationType {
+  case class StartsWith(literal: String) extends RegexOptimizationType
+  case class Contains(literal: String) extends RegexOptimizationType
+  case class PrefixRange(literal: String, charsRange: (Int, Int), length: Int) 
```

Comment:
The title of this says that it supports `literal[a-b]{x,y}`, but this only supports `literal[a-b]{x}`. We need to update one or both of these.

Reply:
It is intended to check if a string contains (instead of matches) `literal[a-b]{x,y}`, so if there are already x characters satisfying [a-b], then the result will be true no matter what follows. 

For example, "abcabc" is not a match for the pattern `^[a-c]{2,3}$`, but it does contain a `[a-c]{2,3}`, and we know that already after checking the first two characters.